packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +401,536 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+  ///
+  /// The [controller] property must be set in this case.
+  /// It should be passed the relevant [Scrollable]'s [ScrollController].
+  ///
+  /// Defaults to false.
+  ///
+  /// {@tool snippet}
+  ///
+  /// ```dart
+  /// final ScrollController _controllerOne = ScrollController();
+  /// final ScrollController _controllerTwo = ScrollController();
+  ///
+  /// build(BuildContext context) {
+  /// return Column(
+  ///   children: <Widget>[
+  ///     Container(
+  ///        height: 200,
+  ///        child: Scrollbar(
+  ///          isAlwaysShown: true,
+  ///          controller: _controllerOne,
+  ///          child: ListView.builder(
+  ///            controller: _controllerOne,
+  ///            itemCount: 120,
+  ///            itemBuilder: (BuildContext context, int index)
+  ///                => Text('item $index'),
+  ///          ),
+  ///        ),
+  ///      ),
+  ///      Container(
+  ///        height: 200,
+  ///        child: CupertinoScrollbar(
+  ///          isAlwaysShown: true,
+  ///          controller: _controllerTwo,
+  ///          child: SingleChildScrollView(
+  ///            controller: _controllerTwo,
+  ///            child: SizedBox(height: 2000, width: 500,),
+  ///          ),
+  ///        ),
+  ///      ),
+  ///    ],
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  final bool isAlwaysShown;
+
+  /// [Radius] of corners if the scrollbar should have rounded corners.
+  ///
+  /// Scrollbar will be rectangular if [radius] is null, which is the default
+  /// behavior.
+  final Radius? radius;
+
+  /// The thickness of the scrollbar in the cross axis of the scrollable.
+  ///
+  /// Cannot be null.
+  final double thickness;
+
+  /// The color of the scrollbar thumb.
+  ///
+  /// Cannot be null.
+  final Color thumbColor;
+
+  /// The [Duration] of the fade animation.
+  ///
+  /// Cannot be null, defaults to a [Duration] of 300 milliseconds.
+  final Duration fadeDuration;
+
+  /// The [Duration] of time until the fade animation begins.
+  ///
+  /// Cannot be null, defaults to a [Duration] of 600 milliseconds.
+  final Duration timeToFade;
+
+  /// The [Duration] of time that a LongPress will trigger the drag gesture of
+  /// the scrollbar thumb.
+  ///
+  /// Cannot be null, defaults to [Duration.zero].
+  final Duration pressDuration;
+
+  @override
+  RawScrollbarState<RawScrollbar> createState() => RawScrollbarState<RawScrollbar>();
+}
+
+/// The state for a [RawScrollbar] widget, also shared by the [Scrollbar] and
+/// [CupertinoScrollbar] widgets.
+///
+/// Controls the animation that fades a scrollbar thumb in and out of view.
+class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProviderStateMixin<T> {
+  double? _dragScrollbarAxisPosition;
+  Drag? _drag;
+  ScrollController? _currentController;
+
+  /// A [GlobalKey] associated with the [painter], will be assigned to the
+  /// [CustomPaint] of the scrollbar.
```

Comment:
What's this used for exactly (maybe it needs a more use-specific name) and what _must_ subclasses do?